test/Microsoft.AspNetCore.Razor.Tools.Test/Roslyn/BuildProtocolTest.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CommandLine;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Tools
+{
+    public class BuildProtocolTest
+    {
+        [Fact]
+        public async Task ReadWriteCompleted()
+        {
+            // Arrange
+            var response = new CompletedBuildResponse(42, utf8output: false, output: "a string");
```

Comment:
We should think about renaming the request and response classes. We don't really 'build', we do other things.

Reply:
I was actually thinking of renaming these to `ServerProtocol`, `ServerRequest` and `ServerResponse`. But that will be in a separate cleanup PR.